org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/PreferenceManager.java

```diff
@@ -164,6 +164,7 @@ public static void initializeJavaCoreOptions() {
 		// workaround for https://github.com/redhat-developer/vscode-java/issues/718
 		javaCoreOptions.put(JavaCore.CORE_CIRCULAR_CLASSPATH, JavaCore.WARNING);
 		javaCoreOptions.put(JavaCore.COMPILER_IGNORE_UNNAMED_MODULE_FOR_SPLIT_PACKAGE, JavaCore.ENABLED);
+		javaCoreOptions.put(JavaCore.COMPILER_PB_NULL_UNCHECKED_CONVERSION, JavaCore.IGNORE);
```

Comment:
Wouldn't it make more sense to put it within the if/else at https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/2016a78a03ec3f26ba857e81d72fe4396e1b05a3/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java#L2498 ? Otherwise, looks mostly fine.